pages/_app.js

```diff
@@ -1,10 +1,35 @@
+import Navigation from "@/components/Navigation";
 import GlobalStyle from "../styles";
+import { useState } from "react";
+import plants from "@/assets/plants";
 
 export default function App({ Component, pageProps }) {
+  const [ownedPlants, setOwnedPlants] = useState([]);
```

Comment:
Since your state only collects `id`s i recommend to name it 'ownedPlantIds'. 
Being precise with naming helps a lot to easily understand code.